homeassistant/components/pterodactyl/entity.py

```diff
@@ -0,0 +1,42 @@
+"""Base entity for the Pterodactyl integration."""
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import PterodactylCoordinator
+
+MANUFACTURER = "Pterodactyl"
+
+
+class PterodactylEntity(CoordinatorEntity[PterodactylCoordinator]):
+    """Representation of a Pterodactyl base entity."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: PterodactylCoordinator,
+        identifier: str,
+        config_entry: ConfigEntry,
+    ) -> None:
+        """Initialize base entity."""
+        super().__init__(coordinator)
+
+        index = coordinator.api.get_index_from_identifier(identifier)
+
+        if index is None:
+            raise HomeAssistantError(
+                f"Identifier '{identifier}' not found in data list"
+            )
```

Comment:
We should not raise home assistant errors here. We should only create entities for things we know exist.

Same for the is_on in binary sensor